cmds/upgraded/main.go

```diff
@@ -188,39 +177,43 @@ func main() {
 		log.Info().Msg("shutting down")
 	})
 
-	// 4. Start watcher for new version
 	if bootMethod != upgrade.BootMethodFList {
+		log.Info().Msg("node is not booted from an flist. upgrade is not supported")
 		<-ctx.Done()
-	} else {
-		log.Info().Msg("start upgrade daemon")
-		ticker := time.NewTicker(time.Second * time.Duration(interval))
-
-		for {
-			err := SafeUpgrade(&upgrader)
-			if err == upgrade.ErrRestartNeeded {
-				log.Info().Msg("restarting upgraded")
-				return
-			} else if err != nil {
-				//TODO: crash or continue!
-				log.Error().Err(err).Msg("upgrade failed")
-			}
-
-			version, err := upgrader.Version()
-			if err != nil {
-				log.Fatal().Err(err).Msg("failed to read current version")
-			}
-
-			log.Info().Str("version", version.String()).Msg("new version installed")
-
-			if _, err = idStore.RegisterNode(nodeID, farmID, version.String()); err != nil {
-				log.Error().Err(err).Msg("fail to register node identity")
-			}
-
-			select {
-			case <-ticker.C:
-			case <-ctx.Done():
-				break
-			}
+		return
+	}
+
+	// 4. Start watcher for new version
+	log.Info().Msg("start upgrade daemon")
+	ticker := time.NewTicker(time.Second * time.Duration(interval))
+	defer ticker.Stop()
+
+	for {
+		err := SafeUpgrade(&upgrader)
+		if err == upgrade.ErrRestartNeeded {
+			log.Info().Msg("restarting upgraded")
+			return
+		} else if err != nil {
+			//TODO: crash or continue!
+			log.Error().Err(err).Msg("upgrade failed")
+		}
+
+		version, err := upgrader.Version()
+		if err != nil {
+			log.Fatal().Err(err).Msg("failed to read current version")
+		}
+
+		log.Info().Str("version", version.String()).Msg("current")
+
+		if _, err = idStore.RegisterNode(nodeID, farmID, version.String()); err != nil {
```

Comment:
Maybe we should also have a backoff retry here in case there is a network issue that prevent to register the version directly ?

Reply:
Idd, will add it